qiskit/quantum_info/operators/unitary.py

```
@@ -0,0 +1,250 @@
+                self.matrix = numpy.linalg.matrix_power(
+                    self.matrix.T.conj(), n)
+            return self
+        else:
+            dim = self.dimension
+            uni = Unitary(numpy.empty((dim, dim), dtype='complex'), validate=False)
+            if n >= 0:
+                uni.matrix = numpy.linalg.matrix_power(self.matrix, n)
+            else:
+                uni.matrix = numpy.linalg.matrix_power(
+                    self.matrix.T.conj(), n)
+            return uni
+
+    @matrix.setter
+    def matrix(self, mat):
+        """set matrix representation
+
+        Args:
+            mat (ndarray | list(list)): matrix to set
+
+        Raises:
+            QiskitError: if matrix is not unitary
+        """
+        if mat is not None:
+            mat = numpy.asarray(mat, dtype='complex')
+            if self._validate:
+                if not numpy.allclose(mat.T.conj() @ mat, numpy.identity(mat.shape[0]),
+                                      rtol=self._rtol, atol=self._atol):
+                    raise QiskitError("matrix is not unitary")
+            self._matrix = mat
+            self._n_qubits = int(numpy.log2(mat.shape[0]))
+            self._dimension = mat.shape[0]
+        else:
+            self._matrix = None  # for creating empty Unitary
+
+    def reapply(self, circ):
+        """reapply this gate to circ; TODO: remove when monkey patching removed"""
+        self._modifiers(circ._attach(self))
+
+    def _define_decompositions(self):
+        decomposition = DAGCircuit()
+        self._decompositions = [decomposition]
+
+    @property
+    def label(self):
+        """get unitary label
+
+        Returns:
+            str: object's label
+        """
+        return self._label
+
+    @label.setter
+    def label(self, name):
+        """set unitary label to name
+
+        Args:
+            name (str): label to assign unitary
+        """
+        if isinstance(name, str):
+            self._label = name
```

this will silently not do anything if a non-string label is passed. If you always expect a string, then maybe should raise an error?

---

corrected, thanks